modules/ROOT/pages/policy-openid-connect.adoc

```diff
@@ -8,98 +8,182 @@ endif::[]
+}
+---
+
+The fields returned by the OAuth provider are processed by the policy, propagated throughout the Mule flow, and finally exposed to the backend if the application requesting the access uses an HTTP requester.
+
+=== Token Caching
 
-[listing]
-anypoint.platform.max_federation_expiration_time=<expiration time in seconds; specify 0 to disable caching>
+After the system validates a token, it is cached (by default) until it expires, to improve the performance of the policy. In certain cases, such as in the case of revoked tokens, you might want to minimize or even disable caching.
 
-== Communicating to the OAuth provider through a proxy
+To control the token-cached time before retrying a validation against the token validation endpoint, specify the following property when starting the Mule runtime engine:
 
-To enable the OpenID Connect OAuth 2.0 Token Enforcement Policy to use the Gateway proxy settings, specify the following property:
+`anypoint.platform.max_federation_expiration_time=<expiration time in seconds; specify 0 to disable caching>`
 
-[listing]
-anypoint.platform.external_authentication_provider_enable_proxy_settings=<true|false(default)>
+Alternatively, you can also specify this parameter in the `wrapper.conf` file.
+ 
+To control the number of tokens that can be cached simultaneously, specify the following property when starting the Mule runtime engine:
+ 
+`anypoint.platform.openid_connect_cache_max_size=<amount of tokens to be cached; default value is 10.000>`
 
-Setting the previous property to true makes the policy use the runtime proxy settings if specified. For example:
+Alternatively, you can also specify this parameter in the `wrapper.conf` file.
 
-----
-anypoint.platform.proxy_host=localhost
-anypoint.platform.proxy_port=8080
-----
+[[token-validation-endpoint-authentication]]
+=== Token Validation Endpoint Authentication
 
-== Leveraging OAuth provider information
+To complete validating the provided token, the policy sends a request to the validation endpoint. The policy then obtains the credentials from the client provider configuration (Token Introspection Client section) in Access management. These credentials are sent in the request body, as specified in the https://tools.ietf.org/html/rfc6749#section-2.3[Client Authentication section of the OAuth 2.0 Authorization Framework specification]. 
 
-When a token is successfully validated by the OAuth provider, the provider sends back to the policy who requested the token validation, some extra information, including fields that are OpenID Connect side configurable.
+Other authentication methods for the token validation endpoint are currently not supported.
 
-Token validation response example from the authorization server:
+== Configuring the Expose Headers Option
+
+If you configured the `Expose Headers` option in the policy, the application (for example, out-of-the-box proxies) uses the HTTP requester to redirect the properties to the backend as headers. 
```

Comment:
Is "(for example, out-of-the-box proxies)" from a previous topic?
In general, we want to keep topics independent, so that updating them does not create reference errors.
If you use one example for several topics, be sure to make that clear in the context, and call it "the example in <name topic>" or some other identifier besides "previous topic/section/whatever."
See the style ref. entry for "self-reference."